docs/ssh-tutorial.md

```diff
@@ -0,0 +1,138 @@
+# Tutorial
+
+Create a reverse proxy to allow SSH access to a computer
+
+Scenario: You want to allow SSH access to a computer that doesn't have a public IP, is inside a private network or behind
+a firewall
+
+
+
+## Setup your exit node
+
+#### Using inletsctl
+You will need to have an account and API key with one of the [supported providers](https://github.com/inlets/inletsctl#featuresbacklog), 
+we are using Digital Ocean in this example. So download your access key to a file called `do-access-token` and put it in
+your home directory.
+
+
+First of all, download `inletsctl` to your local machine
+```sh
+# Install Inletsctl to /usr/local/bin
+curl -sLSf https://inletsctl.inlets.dev | sudo sh
+
+# Create an exit node
+
+inletsctl create --access-token-file ~/do-access-token --remote-tcp 127.0.0.1
+```
+This should then create an exit node, which will forward all TCP traffic to through the client to the `remote-tcp` address
+that we set, in this case `127.0.0.1`
+
+We should also see an outputted command that we can use on the machine we want to ssh onto.
+
+
+```sh 
+  export TCP_PORTS="8000"
+  export LICENSE=""
+  inlets-pro client --connect "wss://206.189.114.179:8123/connect" \
+        --token "4NXIRZeqsiYdbZPuFeVYLLlYTpzY7ilqSdqhA0HjDld1QjG8wgfKk04JwX4i6c6F" \
+        --license "$LICENSE" \
+        --tcp-ports $TCP_PORTS
+
+```
+
+
+#### Manually
+Provision a VM on DigitalOcean or another IaaS provider.
+
+Log in with ssh and obtain the binary:
+
+```sh
+curl -SLsf https://github.com/inlets/inlets-pro-pkg/releases/download/0.4.3/inlets-pro > inlets-pro
+chmod +x ./inlets-pro
+mv ./inlets-pro /usr/bin/inlets-pro
+```
+
+Find your public IP:
+
+```
+export IP=$(curl -s ifconfig.co)
+```
+
+Confirm the IP with `echo $IP` and save it, you need it for the client
+
+Get an auth token and save it for later to use with the client
+
+```sh
+export TOKEN="$(head -c 16 /dev/urandom |shasum|cut -d'-' -f1)"
+
+echo $TOKEN
+```
+
+Start the server:
+
+```sh
+sudo inlets-pro server \
+  --auto-tls \
+  --common-name $IP \
+  --remote-tcp 127.0.0.1 \
+  --token $TOKEN
+```
+
+
+## Configure your ssh agent
+(port 22 conflict with exit-node)
+
+Our cloud exit node is already using port 22 for SSH access. This means we need to configure our machine to listen
+for ssh connections on another port.
+
+If we find the line in  `/etc/ssh/sshd_config` near the top that is `Port 22`, make sure it's un-commented
+and add `Port 2222`
+
+```
+Port 22
+Port 2222
+```
+
+Next we recommend disabling password authentication, and instead relying on SSH Keys only.
+
+> WARNING: make sure you have validated that SSH works using your SSH Key before disabling password authentication.
```

Comment:
Maybe suggest ssh-copy-id

Reply:
Wow i have been doing this manually for years.